sdk/core/core-https/documentation/core2.md

```diff
@@ -0,0 +1,69 @@
+# Azure SDK for JS Core 2.0
+
+## What do you mean 2.0?
+
+Today one of the primary core packages we ship is `@azure/core-http`. As of this document being written it sits at version 1.1.5. Due to a number factors discussed in this document, there is a desire to make some breaking changes that would create a new major version of this package (i.e. "2.0.0".)
+
+Since `@azure/core-http` is intended only for indirect use by most consumers, a major version change would not be largely disruptive. Currently, in order to preserve compatibility and not complicate the story for servicing, a new package has been created that is currently named `@azure/core-https`. This name is not final.
+
+## Motivations, Context, and History
+
+In ancient times (~5 years ago), Microsoft provided SDKs for NodeJS developers wanting to interact with Azure services. Ultimately, many of these packages were generated using [AutoRest](https://github.com/Azure/Autorest) using a package called `autorest.nodejs`. The autogenerated code had the need for a common, re-usable core for performing HTTP operations, and thus [`ms-rest`](https://github.com/Azure/ms-rest-js) was created.
+
+Besides `ms-rest` for common code, another package `ms-rest-azure` was created to house Azure-specific components and authetication code that worked with credentials. At this point, support for TypeScript came in the form of hand-authored d.ts files.
+
+Later, there emerged a desire for isomorphic packages, packages that worked in both the browser as well as Node. As part of this effort, code was to be generated in TypeScript directly and a new repo was created: `azure-sdk-for-js`.
+
+A new code generator was created (`autorest.typescript`) that replaced the existing one (`autorest.nodejs`.) `ms-rest` became `@azure/ms-rest-js`. The authentication pieces from `@ms-rest-azure` were put into `@azure/ms-rest-nodeauth` and an equivalent package was created for the browser called `@azure/ms-rest-browserauth`. All remaining parts of `ms-rest-azure` were placed into `@azure/ms-rest-azure-js`.
+
+`@azure/ms-rest-js` was used as the basis for `@azure/core-http` when the JS team for the new Azure SDK was formed. This was nice as it bootstrapped the ability to make HTTP requests and provided broad compatibility with many track 1 clients that still exist in the `@azure/sdk-for-js` repository today.
```

Comment:
Are we trying to say that `core-http` can be used with Track 1 clients?

Reply:
I think so? Don't we? Or do those only rely on `ms-rest-js`?

Reply:
Track 1 clients are all auto generated using v4 of the code generator and use `@azure/ms-rest-js` and`@azure/ ms-rest-nodeauth`
If they use v5 of the code generator, then will be using `@azure/core-http` and `@azure/identity`

Reply:
Hmm so should we say that it was *originally* designed for interop, but we have since relaxed that as a requirement?

Reply:
So, this section is about the benefits of using ms-rest-js as the basis for core-http.

The only benefit worth calling out is that the move from Track 1 to Track 2 for auto generated libraries is made easier by relying on core-http which was almost the same as ms-rest-js. The other one is that ms-rest-js was a tried and tested, and we had no reason at the time to create anything from scratch